lib/identity_cache/query_api.rb

```diff
@@ -468,7 +485,23 @@ def prefetch_one_association(association, records)
             prefetch_embedded_association(records, association, details)
             parent_records = records.map(&details[:cached_accessor_name].to_sym).compact
           else
-            raise ArgumentError.new("Non-embedded has_one associations do not support prefetching yet.")
+            ids_to_parent_record = records.each_with_object({}) do |record, hash|
+              child_ids = Array(record.send(details[:cached_id_name]))
+              child_ids.each do |child_id|
+                hash[child_id] = record
+              end
+            end
+
+            parent_record_to_child_records = Hash.new { |h, k| h[k] = [] }
+            child_records = details[:association_reflection].klass.fetch_multi(*ids_to_parent_record.keys)
+            child_records.each do |child_record|
+              parent_record = ids_to_parent_record[child_record.id]
+              parent_record_to_child_records[parent_record] << child_record
+            end
+
+            parent_record_to_child_records.each do |parent, children|
+              parent.send(details[:prepopulate_method_name], children)
```

Comment:
This is wrong, it is preloading an array for a cache_has_one association.  This indicates that we are also missing a test for this.